### PR TITLE
Feature/sandbox service apis update

### PR DESCRIPTION
**Pull request by vgonzalez9**
Fix some of UI Integration fixes
Added missing @JsonProperty that was accidentally removed.
Added customer deserializers for Set and List in ReportingSystemSettings & AggregateReportingProperties to handle a single item to convert to a list
Fix the PUT for the tenantService - it's checking tenantProperties when that isn't ready to be checked
Fix the PUT for SandboxService - it was using tenantKey for the save and sandbox.key
Also for a sandbox create, the id & key  are automatically generated, using the tenantId and a string version of the currentTime in ms

### CODE REVIEW

File: admin-service/src/main/java/org/opentestsystem/rdw/admin/model/ApplicationSandboxConfiguration.java [hunk 1]
```diff
@@ -11,12 +12,12 @@
     /**
      * Default Reporting system properties
      */
+    @JsonProperty("reporting")
```
Comment (malaffoon) [inline admin-service/src/main/java/org/opentestsystem/rdw/admin/model/ApplicationSandboxConfiguration.java]: I thought it used the variable name by default; just curious what it was calling the property?

Comment (vgonzalez9) [reply admin-service/src/main/java/org/opentestsystem/rdw/admin/model/ApplicationSandboxConfiguration.java]: its using the class name

File: admin-service/src/main/java/org/opentestsystem/rdw/admin/service/impl/DefaultSandboxService.java [hunk 2]
```diff
@@ -88,4 +93,13 @@ private void validateExistingSandbox(final SandboxConfiguration sandboxConfigura
...
         }
     }
+
+    /**
+     * TODO Improve
+     * @return a tenant based ID - using current time
+     */
+    private String generateId(String tenantId) {
+        String ts = String.valueOf(System.currentTimeMillis());
+        return tenantId + ts;
```
Comment (malaffoon) [inline admin-service/src/main/java/org/opentestsystem/rdw/admin/service/impl/DefaultSandboxService.java]: perhaps a 4-6 character hash of the timestamp?

File: common/src/main/java/org/opentestsystem/rdw/reporting/common/configuration/AggregateReportingPropertiesTenant.java [hunk 3]
```diff
@@ -1,13 +1,19 @@
...
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+
 import java.util.Set;
 
 public class AggregateReportingPropertiesTenant implements AggregateReportingProperties {
 
+    @JsonFormat(with = JsonFormat.Feature.ACCEPT_SINGLE_VALUE_AS_ARRAY)
     private Set<String> assessmentTypes;
 
+    @JsonDeserialize(using = CustomSetStringDeserializer.class)
```
Comment (burzum619) [inline common/src/main/java/org/opentestsystem/rdw/reporting/common/configuration/AggregateReportingPropertiesTenant.java]: Shouldn't all of these properties have the same annotations?

File: admin-service/src/main/java/org/opentestsystem/rdw/admin/model/ApplicationSandboxConfiguration.java (hunk 1 above)
Comment (burzum619) [reply admin-service/src/main/java/org/opentestsystem/rdw/admin/model/ApplicationSandboxConfiguration.java]: I think by default its using property/variable names.

File: common/src/main/java/org/opentestsystem/rdw/reporting/common/configuration/AggregateReportingPropertiesTenant.java (hunk 3 above)
Comment (vgonzalez9) [reply common/src/main/java/org/opentestsystem/rdw/reporting/common/configuration/AggregateReportingPropertiesTenant.java]: I created 2 different serializers one for List and one for Set. Or are you suggesting the whole class should be using the same one?

File: admin-service/src/main/java/org/opentestsystem/rdw/admin/model/ApplicationSandboxConfiguration.java (hunk 1 above)
Comment (vgonzalez9) [reply admin-service/src/main/java/org/opentestsystem/rdw/admin/model/ApplicationSandboxConfiguration.java]: And you both are correct that annotation is not needed and the name is use by default.

File: common/src/main/java/org/opentestsystem/rdw/reporting/common/configuration/AggregateReportingPropertiesTenant.java (hunk 3 above)
Comment (vgonzalez9) [reply common/src/main/java/org/opentestsystem/rdw/reporting/common/configuration/AggregateReportingPropertiesTenant.java]: Thanks simplified how this is being done, and now properly handling the effectiveLanguages List.

**Pull request merged**